lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+  alias Entice.Logic.Npc
+
+
+  defstruct(
+    players: [],
+    npcs: [], #TODO: Figure out if we want the npcs ot be able to join&leave or stay there even if dead or unspawned
+    map: nil)
+
+  @doc """
+  Registers a MapInstance.Behaviour for the Entity.
+  Params:
```

Comment:
From the elixir agent.ex source code:
```Elixir
  @doc """
  Stops the agent with the given `reason`.
  It returns `:ok` if the server terminates with the given
  reason, if it terminates with another reason, the call will
  exit.
  This function keeps OTP semantics regarding error reporting.
  If the reason is any other than `:normal`, `:shutdown` or
  `{:shutdown, _}`, an error report will be logged.
  """
  @spec stop(agent, reason :: term, timeout) :: :ok
  def stop(agent, reason \\ :normal, timeout \\ 5_000) do
```
Note the typespec definition above the `def` - i dont think we need to go through extensive lenghts to document our code this way and place typespecs everywhere... i actually think that would be kind of problematic since it clutters the code ;)
But if you wanna do something like that, check the original conventions ;)